src/Microsoft.Diagnostics.Monitoring.HostingStartup/ParameterCapturing/FunctionProbes/FunctionProbesStub.cs

```diff
@@ -28,6 +29,12 @@ public static void EnterProbeStub(ulong uniquifier, object[] args)
                 return;
             }
 
+            // Don't probe our own code.
+            if (MonitorExecutionContextTracker.IsInMonitorContext())
```

Comment:
We'll want to move this check after `s_inProbe = true`  since the call to `MonitorExecutionContextTracker.IsInMonitorContext()` could indirectly call into another function with probes in it.